### PR TITLE
SIMSBIOHUB-623: Add Site Name and Description When Creating New Sampling Site

### PR DESCRIPTION
**Pull request by mauberti-bc**
## Links to Jira Tickets

- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-623

## Description of Changes

- Allows a name and description to be added to a sampling site during creation

## Testing Notes

- Can only name/describe a sampling site during creation when exactly one site is being added

### CODE REVIEW

File: app/src/features/surveys/sampling-information/sites/create/form/SampleSiteGeneralInformationCreateForm.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import Grid from '@mui/material/Grid';
+import TextField from '@mui/material/TextField';
+import { useFormikContext } from 'formik';
+import { ChangeEvent } from 'react';
+
+/**
+ * Component for adding a name and description to a sampling site on the create form.
+ *
+ * NOTE: When creating a sampling site, the sites are an array and the name and description apply to each of the sites in the array.
+ * A name and description can only be manually entered when creating a single sampling site. If creating multiple,
```
Comment (MacDeluca) [inline app/src/features/surveys/sampling-information/sites/create/form/SampleSiteGeneralInformationCreateForm.tsx:11]: question: Is this interaction intended? Or do we want to be able to change the sampling site name and description for all?

Comment (mauberti-bc) [reply app/src/features/surveys/sampling-information/sites/create/form/SampleSiteGeneralInformationCreateForm.tsx:11]: When drawing, you can only add one site per form submission. Renaming multiple in one form submission is ideal, but we'd need to allow for multiple sites per submission first. 

The upcoming new survey area control could be reused once it gets merged for this - it allows multiple features to be renamed.

**Comment by MacDeluca** [comment]
Looks good!

Only minor thing I noticed is the difference of the sampling site description input and the technique description. Might be nice to include the optional / max character count for the sampling site description too.

Sampling Site description:
image

Technique description:
image

**Review by MacDeluca — APPROVED** [review]
🍴

**Pull request merged**